linode/apl-core
# feat: oauth2-proxy custom error page

### Implements
https://jira.linode.com/browse/APL-209

### Description
This PR adds custom error page for oauth2-proxy and shows the user a loading page (see screenshots below) and redirects them to the console dashboard.

### Testing

1. Create a cluster from `APL-209` branch.
2. Open a tab (preferably incognito tab) and navigate to the console URL (console.`<domainSuffix>`).
3. When the Keycloak login page appears, do not log in.
4. Open a second tab and navigate to the console URL again in the new tab.
5. When the Keycloak login page appears in the second tab, do not log in.
6. Go back to the first tab and attempt to log in with your credentials.
7. Verify that the loading page is instantly visible (or not) and redirect to the console dashboard.
8. Reduce network speed to the `3G` (see screenshots below) from the developer tools network tab for better testing 
(Don't forget to reset (to the `No throttling`) after testing).
9. When the loading page appears (with `3G` speed), stop the browser loading by clicking the `X` button next to the address bar.
10. Verify that the manual redirect works by clicking the `click here` button in the loading page.

### Screenshots
Oauth2-Proxy Custom Error Page (Loading&Redirect Page)
![oauth2-custom-error-page](https://github.com/user-attachments/assets/0f8f0e36-0b41-48e6-8f2e-329448d0988e)

Reduce Network Speed
![Screenshot 2024-10-14 at 09 32 13](https://github.com/user-attachments/assets/45bb19d1-94fa-4d9a-a163-ade8689cd236)

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.
